app/services/re_captcha/check_enabled.rb

```diff
@@ -0,0 +1,37 @@
+# This service encapsulates the logic related to validating if reCAPTCHA is
+# enabled in the current Forem instance. The decision is based on making
+# sure the necessary SiteConfig keys are available and also on the user
+# object passed in.
+#
+# Example use: ReCaptcha::CheckEnabled.call(current_user) => true/false
+module ReCaptcha
+  class CheckEnabled
+    def self.call(current_user = nil)
+      new(current_user).call
+    end
+
+    def initialize(current_user)
+      @current_user = current_user
+    end
+
+    def call
+      # recaptcha will not be enabled if site key and secret key aren't set
+      return false unless keys_configured?
+      # recaptcha will always be enabled when not logged in
+      return true if @current_user.nil?
+      # recaptcha will not be enabled for trusted/admin/tag mod users
+      return false if @current_user.auditable?
```

Comment:
**Suggestion**: explicitly spell out which roles you do or don't mean here. `.auditable?` strikes me as a way of asking "does the current user have admin rights?" or "is the current user just a regular user?"

Reply:
Thanks for the suggestion! I added it because it will definitely make it more clear to understand when reading the code